src/SchemaRegistry.ts

```diff
@@ -165,39 +166,50 @@ export default class SchemaRegistry {
   }
 
   private async updateOptionsWithSchemaReferences(
-    options: SchemaRegistryAPIClientOptions | undefined,
     schema: ConfluentSchema,
+    options: SchemaRegistryAPIClientOptions | undefined,
```

Comment:
No reason to mark type as `| undefined` - simply mark it is a optional, as that is more correct for the purpose of it. I do not think you would expect users to do:

```
await updateOptionsWithSchemaReferences(schema, undefined);
```

but rather do

```
await updateOptionsWithSchemaReferences(schema);
```